tests/core/full_node/test_mempool.py

```diff
@@ -1878,3 +1932,203 @@ def test_parse_unknown_condition(self):
+        # CREATE_COIN
+        puzzle_hash = "abababababababababababababababab"
+
+        # this max cost is exactly enough for the create coin condition
+        npc_result = generator_condition_tester(
+            f'(51 "{puzzle_hash}" 10) ', max_cost=20470 + 95 * COST_PER_BYTE + 1800000
+        )
+        assert npc_result.error is None
+        assert npc_result.clvm_cost == 20470
+        assert len(npc_result.npc_list) == 1
+
+        # if we subtract one from max cost, this should fail
+        npc_result = generator_condition_tester(
+            f'(51 "{puzzle_hash}" 10) ', max_cost=20470 + 95 * COST_PER_BYTE + 1800000 - 1
+        )
+        assert npc_result.error in [Err.BLOCK_COST_EXCEEDS_MAX.value, Err.INVALID_BLOCK_COST.value]
+
+    def test_agg_sig_cost(self):
+        # AGG_SIG_ME
+        pubkey = "abababababababababababababababababababababababab"
+
+        # this max cost is exactly enough for the AGG_SIG condition
+        npc_result = generator_condition_tester(
+            f'(49 "{pubkey}" "foobar") ', max_cost=20512 + 117 * COST_PER_BYTE + 1200000
+        )
+        assert npc_result.error is None
+        assert npc_result.clvm_cost == 20512
+        assert len(npc_result.npc_list) == 1
+
+        # if we subtract one from max cost, this should fail
+        npc_result = generator_condition_tester(
+            f'(49 "{pubkey}" "foobar") ', max_cost=20512 + 117 * COST_PER_BYTE + 1200000 - 1
+        )
+        assert npc_result.error in [Err.BLOCK_COST_EXCEEDS_MAX.value, Err.INVALID_BLOCK_COST.value]
+
+    def test_create_coin_different_parent(self):
+
+        # if the coins we create have different parents, they are never
+        # considered duplicate, even when they have the same puzzle hash and
+        # amount
+        puzzle_hash = "abababababababababababababababab"
+        program = SerializedProgram.from_bytes(
+            binutils.assemble(
+                f'(q ((0x0101010101010101010101010101010101010101010101010101010101010101 (q (51 "{puzzle_hash}" 10)) 123 (() (q . ())))(0x0101010101010101010101010101010101010101010101010101010101010102 (q (51 "{puzzle_hash}" 10)) 123 (() (q . ()))) ))'  # noqa
+            ).as_bin()
+        )
+        generator = BlockGenerator(program, [])
+        npc_result: NPCResult = get_name_puzzle_conditions(
+            generator, MAX_BLOCK_COST_CLVM, cost_per_byte=COST_PER_BYTE, safe_mode=False
+        )
+        assert npc_result.error is None
+        assert len(npc_result.npc_list) == 2
+        opcode = ConditionOpcode.CREATE_COIN
+        assert npc_result.npc_list[0].conditions == [
+            (
+                opcode.value,
+                [ConditionWithArgs(opcode, [puzzle_hash.encode("ascii"), bytes([10])])],
+            )
+        ]
+        assert npc_result.npc_list[1].conditions == [
+            (
+                opcode.value,
+                [ConditionWithArgs(opcode, [puzzle_hash.encode("ascii"), bytes([10])])],
+            )
+        ]
```

Comment:
Purely cosmetic so not necessary but maybe dedup this

Reply:
fixed